### PR TITLE
test: fix `lit`

### PR DESCRIPTION
**Pull request by userquin**
After some testing it seems there is a problem on `happy-dom` with custom elements and `reactive-element` library.

`reactive-element` has a `DEV_MODE` flag that is initialized to `true` (`const DEV_MODE = true;`), that forces to eval the `localName` of the custom element (line 652 on `node_modules/.pnpm/@lit+reactive-element@1.0.2/node_modules/@lit/reactive-element/development/reactive-element.js`).

Since `happy-dom` uses `this.tagName.toLowerCase()` on the `localName` getter (line 112 on `node_modules/.pnpm/happy-dom@2.25.0/node_modules/happy-dom/lib/nodes/element/Element.js`), then we got the error `TypeError: Cannot read properties of null (reading 'toLowerCase')` because the `tagName` is missing from the custom element.

I will send also a PR to fix it on `happy-dom`.

### CODE REVIEW

**Comment by patak-dev** [comment]
Great finding @userquin! Maybe let's wait a bit and check how fast the happy-dom maintainers react?

**Comment by userquin** [comment]
@patak-dev @antfu I don't know why `this.tagName` is `null`: this code returns the custom component constructor `await customElements.get('my-button')`.

File: test/lit/test/basic.test.ts
Comment (antfu) [inline test/lit/test/basic.test.ts:12]: Can you add a TODO comment linking to the PR you create and we can remove this after it's been merged?

Comment (userquin) [reply test/lit/test/basic.test.ts:13]: uhmm, do you mean the PR to happy-dom? wait I need to make the PR first ubt @patak-dev want to wait...

Comment (patak-dev) [reply test/lit/test/basic.test.ts:13]: @antfu wants to merge it 👍🏼

**Review by patak-dev — APPROVED** [review]

**Pull request merged**
